components/AddHabit.js

```diff
@@ -3,53 +3,37 @@ import EmojiModal from "./EmojiModal";
 import InlineEdit from "./InlineEdit";
 
 export default function AddHabit(props) {
-  const { setHabits, id } = props;
+  const { setHabits, listId } = props;
   const [isFormShowing, setIsFormShowing] = useState(false);
   const [editingValue, setEditingValue] = useState("");
-  const [isEmojiModalShowing, setIsEmojiModalShowing] = useState(false);
+  const [isModalVisible, setIsModalVisible] = useState(false);
   const [emoji, setEmoji] = useState("🏗");
-  const [checkmark, setCheckmark] = useState("/grayCheckMark.svg");
+  const checkmark = "/grayCheckMark.svg";
 
   const handleSubmit = async (habitText) => {
-    // Get data from the form.
     const data = {
-      slug: id,
+      slug: listId,
```

Comment:
I re-named `habit-lists/[id].js` to `habit-lists/[listId].js` `listId` is the name of the slug query parameter